Modules/System Tests/Cryptography Management/src/CertificateRequestTests.Codeunit.al

```diff
@@ -0,0 +1,75 @@
+codeunit 132590 "CertificateRequestTests"
+{
+    Subtype = Test;
+    TestPermissions = Disabled;
+
+    var
+        Assert: Codeunit "Library Assert";
+
+    [Test]
+    procedure CreateCertificateSigningRequest()
+    var
+        CertSigningRequest: Codeunit CertificateRequest;
+        KeyXml: XmlDocument;
+        Root: XmlElement;
+        Node: XmlNode;
+        KeyXmlText, SigningRequestPem : Text;
+        HashAlgorithm: Enum "Hash Algorithm";
+        RSASignaturePadding: Enum "RSA Signature Padding";
+        Oids: List of [Text];
+        BCRTok: Label '-----BEGIN CERTIFICATE REQUEST-----';
+        ECRTok: Label '-----END CERTIFICATE REQUEST-----';
+    begin
+        CertSigningRequest.InitializeRSA(2048, true, KeyXmlText);
+
+        Assert.IsTrue(XmlDocument.ReadFrom(KeyXmlText, KeyXml), 'RSA key is not valid xml data.');
+        Assert.IsTrue(KeyXml.GetRoot(Root), 'Could not get Root element of key.');
+
+        Assert.IsTrue(Root.SelectSingleNode('Modulus', Node), 'Could not find <Modulus> in key.');
+        Assert.IsTrue(Root.SelectSingleNode('DQ', Node), 'Could not find <DQ> in key.');
+
+        CertSigningRequest.InitializeCertificateRequestUsingRSA(
```

Comment:
Lets split this tests into multiple each of them testing a separate interface procedure